tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/annotation/AnnotationMetaEntity.java

```diff
@@ -430,31 +495,72 @@ private String addDaoConstructor(@Nullable ExecutableElement method) {
 		final String sessionVariableName = getSessionVariableName( sessionType );
 		final String name = method == null ? sessionVariableName : method.getSimpleName().toString();
 		final String typeName = element.getSimpleName().toString() + '_';
-		putMember( name,
-				new RepositoryConstructor(
-						this,
-						typeName,
-						name,
-						sessionType,
-						sessionVariableName,
-						dataStore(),
-						context.addInjectAnnotation(),
-						context.addNonnullAnnotation(),
-						method != null,
-						jakartaDataRepository
-				)
-		);
+		
+		if( method == null || !method.isDefault() ) {
```

Comment:
Why was this needed?

Reply:
To support session getters in default methods, as shown in test:

```java
public interface Foo {
 public default Uni<Mutiny.Session> mysession() {
  // custom way to get a session
 }
 @HQL…
}
```

Reply:
Hrm OK, that's interesting. Need to document that since it's a new feature.

Reply:
This is in the PR description ;)

Reply:
Lucky for us it's still there: https://github.com/hibernate/hibernate-orm/pull/7599#issue-2025938172 :)

Let me know if you want me to update the docs.